code/modules/atmospherics/machinery/portable/pipe_scrubber.dm

```diff
@@ -0,0 +1,166 @@
+/obj/machinery/portable_atmospherics/pipe_scrubber
+	name = "pipe scrubber"
+	desc = "A machine for cleaning out pipes of lingering gases. It is a huge tank with a pump attached to it."
+	icon_state = "pipe_scrubber"
+	density = TRUE
+	max_integrity = 250
+	volume = 200
+	///The internal air tank obj of the mech
+	var/obj/machinery/portable_atmospherics/canister/internal_tank
+	///Is the machine on?
+	var/on = FALSE
+	///What direction is the machine pumping to (into scrubber or out to the port)?
+	var/direction = PUMP_IN
+	///the rate the machine will scrub air
+	var/volume_rate = 1000
+	///List of gases that can be scrubbed
+	var/list/scrubbing = list(
+		/datum/gas/plasma,
+		/datum/gas/carbon_dioxide,
+		/datum/gas/nitrous_oxide,
+		/datum/gas/bz,
+		/datum/gas/nitrium,
+		/datum/gas/tritium,
+		/datum/gas/hypernoblium,
+		/datum/gas/water_vapor,
+		/datum/gas/freon,
+		/datum/gas/hydrogen,
+		/datum/gas/healium,
+		/datum/gas/proto_nitrate,
+		/datum/gas/zauker,
+		/datum/gas/halon,
+	)
+
+/obj/machinery/portable_atmospherics/pipe_scrubber/Initialize(mapload)
+	. = ..()
+	internal_tank = new(src)
+	RegisterSignal(internal_tank, COMSIG_QDELETING, PROC_REF(tank_destroyed))
```

Comment:
Probably want COMSIG_ATOM_BREAK instead/in addition to, I don't think most failures actually cause a qdel
Additionally it might be good to a repair act on to the internal canister so it's possible to fix leaks